### PR TITLE
[EOS-10979] Refactor/eos 10979 refactoring credentials

### PR DESCRIPTION
**Pull request by lreciomelero**

### CODE REVIEW

File: pkg/cluster/internal/create/actions/cluster/cluster.go [hunk 1]
```diff
@@ -130,26 +129,39 @@ type Node struct {
 }
 
 type Credentials struct {
-	// AWS
+	AWS              AWSCredentials                `yaml:"aws"`
+	GCP              GCPCredentials                `yaml:"gcp"`
+	GithubToken      string                        `yaml:"github_token"`
+	ExternalRegistry ExternalRegistryCredentials   `yaml:"external_registry"`
+	DockerRegistries []ExternalRegistryCredentials `yaml:"docker_registries"`
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/cluster/cluster.go:136]:
```suggestion
	GithubToken      string                        `yaml:"github_token"`
	DockerRegistries []DockerRegistryCredentials `yaml:"docker_registries"`
```

File: pkg/cluster/internal/create/actions/cluster/cluster.go [hunk 2]
```diff
@@ -130,26 +129,39 @@ type Node struct {
...
+type GCPCredentials struct {
 	ProjectID    string `yaml:"project_id"`
 	PrivateKeyID string `yaml:"private_key_id"`
 	PrivateKey   string `yaml:"private_key"`
 	ClientEmail  string `yaml:"client_email"`
 	ClientID     string `yaml:"client_id"`
 }
 
-type ExternalRegistry struct {
+type ExternalRegistryCredentials struct {
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/cluster/cluster.go:154]:
```suggestion
type DockerRegistryCredentials struct {
```

**Review by stg-0 — CHANGES_REQUESTED** [review]

File: pkg/cluster/internal/create/actions/createworker/createworker.go [hunk 3]
```diff
@@ -34,31 +34,48 @@ type action struct {
...
+// }
+
+// type GCPCredentials struct {
+// 	ProjectID    string `yaml:"project_id"`
+// 	PrivateKeyID string `yaml:"private_key_id"`
+// 	PrivateKey   string `yaml:"private_key"`
+// 	ClientEmail  string `yaml:"client_email"`
+// 	ClientID     string `yaml:"client_id"`
+// }
+
```
Comment (kahun) [inline pkg/cluster/internal/create/actions/createworker/createworker.go:78]: Shall we remove these block comments?

**Review by kahun — APPROVED** [review]
LGTM

File: pkg/cluster/internal/create/actions/cluster/cluster.go (hunk 1 above)
Comment (lreciomelero) [reply pkg/cluster/internal/create/actions/cluster/cluster.go]: resolved

File: pkg/cluster/internal/create/actions/cluster/cluster.go (hunk 2 above)
Comment (lreciomelero) [reply pkg/cluster/internal/create/actions/cluster/cluster.go]: resolved

**Pull request merged**